code/controllers/subsystem/persistence.dm

```diff
@@ -236,6 +275,16 @@ SUBSYSTEM_DEF(persistence)
 	fdel(json_file)
 	WRITE_FILE(json_file, json_encode(file_data))
 
+/datum/controller/subsystem/persistence/proc/CollectFrames()
+	if(LAZYLEN(GLOB.persist_frames)) //So maps without frames don't automatically clear the list
+		for(var/F in GLOB.persist_frames)
+			SaveFrame(F)
+		var/json_file = file("data/npc_saves/frames.json")
+		var/list/file_data = list()
+		file_data["data"] = saved_frames
+		fdel(json_file)
+		WRITE_FILE(json_file, json_encode(file_data))
```

Comment:
these 2 lines => text2file(json_encode(file_data, json_file)

Reply:
the fdel as well?

Reply:
Also this is the Persistence "standard" for writing files

Reply:
code\controllers\subsystem\persistence.dm:285:error: text2file: expected 2 arguments (found 1)